pyfr/integrators/base.py

```diff
@@ -130,6 +134,32 @@ def cfgmeta(self):
         else:
             return {'config': cfg, 'config-0': cfg}
 
+    def grad_pvars(self):
+        try:
+            # Fetch the corrected gradients only if needed
+            grad_soln = self.grad_soln
+        except NotImplementedError:
+            # If the considered solver does not explictly compute gradients
+            # of primitive variables, then raise error
+            raise RuntimeError(f'Solver {self.system.name} not compatible with corrected gradients computation')
+
+        grads_eles = []
+        for soln, grad_soln in zip(self.soln, self.grad_soln):
+            grads_ele = []
+            # Subset and transpose the solution
+            soln = soln.swapaxes(0, 1)
+
+            # Subset and transpose the solution gradients
+            grad_soln = grad_soln.swapaxes(0, 2).swapaxes(1, 2)
```

Comment:
I think this might be equivalent to a rollaxis?

Reply:
I'll have a look at it, I'm not very familiar with `numpy` tbh

Reply:
```
In [1]: import numpy as np

In [2]: x = np.random.randn(10, 20, 30)

In [3]: np.allclose(x.swapaxes(0, 2).swapaxes(1, 2), np.rollaxis(x, 2))
Out[3]: True
```

However, this code should probably be moved out to the plugins themselves (integrate and tavg).

Reply:
Added the `rollaxis` call and also modified `tavg` to use these new gradients